graphz/src/main/scala/coop/rchain/graphz/Graphz.scala

```diff
@@ -10,16 +10,14 @@ trait GraphSerializer[F[_]] {
   def push(str: String, suffix: String = "\n"): F[Unit]
 }
 
-class StringSerializer[F[_]: Applicative: MonadState[?[_], StringBuffer]]
-    extends GraphSerializer[F] {
+class StringSerializer[F[_]: Defer: MonadState[?[_], StringBuffer]] extends GraphSerializer[F] {
   def push(str: String, suffix: String): F[Unit] =
-    MonadState[F, StringBuffer].modify(sb => sb.append(str + suffix))
+    Defer[F].defer(MonadState[F, StringBuffer].modify(sb => sb.append(str + suffix)))
 }
 
-class ListSerializer[F[_]: Applicative: MonadState[?[_], Vector[String]]]
-    extends GraphSerializer[F] {
+class ListSerializer[F[_]: Defer: MonadState[?[_], Vector[String]]] extends GraphSerializer[F] {
   def push(str: String, suffix: String): F[Unit] =
-    MonadState[F, Vector[String]].modify(_ :+ (str + suffix))
+    Defer[F].defer(MonadState[F, Vector[String]].modify(_ :+ (str + suffix)))
```

Comment:
I don't think the `defer`-s are needed here. Unless you've seen this fail with Eval and without the `defer`-s?

Reply:
@ArturGajowy You are right.